### PR TITLE
tests: Add test and patch for issue 2325

### LINKED ISSUE #2325: Rebuilding a `ConditionalDimension` with no factor adds a factor.
If a `ConditionalDimension` is created with `factor=None`, then `self._factor` is set to `None`, but `self.factor` will return 1. As such, when rebuilding a `ConditionalDimension` with no factor, a factor of 1 is added unexpectedly, changing the generated code.

### PR DESCRIPTION
**Pull request by EdCaunt**
Fixes #2325.

### CODE REVIEW

**Review by mloubout — CHANGES_REQUESTED** [review]
I don't really see what's the issue there and why we would need two separate properties. 
If the factor is None it will rebuild with factor=1 that is treated as None so it will rebuild correctly so I don't see where the issue is

**Comment by mloubout** [comment]
I don't see anything wrong on master

```

In [1]: from devito import *

In [2]: x = Dimension(name='x')
   ...: cd = ConditionalDimension(name='cd', parent=x, condition=Ge(x, 1))

In [3]: cd._rebuild()._factor

In [4]: cd._rebuild().factor
Out[4]: 1

In [5]: cd.factor
Out[5]: 1

In [6]: cd._factor


```

File: devito/types/dimension.py [hunk 1]
```diff
@@ -966,6 +966,10 @@ def spacing(self):
 
     @property
     def factor(self):
+        return self._factor
```
Comment (georgebisbas) [inline devito/types/dimension.py:969]: Cant this be `return self._factor or 1`
and drop `def factor_default` ?

**Comment by EdCaunt** [comment]
Looks like this was actually fixed in a separate patch (blame says @mloubout fixed it 5 days ago). Closing.

**Pull request closed without merging**